framework/bundles/org.jboss.tools.rsp.server/src/main/java/org/jboss/tools/rsp/server/model/ServerModel.java

```diff
@@ -272,17 +272,27 @@ private IStatus validateAttributes(IServerType type, Map<String, Object> map) {
 		Set<String> required = util.listAttributes();
 		for (String attrKey : required) {
 			String attributeType = util.getAttributeType(attrKey);
-			validateAttribute(attrKey, map, attributeType , multiStatus);
+			validateAttribute(attrKey, map, attributeType , multiStatus, true);
 		}
+		
+		Attributes optAttr = type.getOptionalAttributes();
+		util = new CreateServerAttributesUtility(optAttr);
+		Set<String> optional = util.listAttributes();
+		for (String attrKey : optional) {
+			String attributeType = util.getAttributeType(attrKey);
+			validateAttribute(attrKey, map, attributeType , multiStatus, false);
+		}
+
```

Comment:
l270-276 and l278-284 are nearly identical. Why not extract a method and replace those blocks with 2 parameterized calls?